### PR TITLE
Add regression test for ML-style generic definitions

### LINKED ISSUE #1868: Wrong generic constraint placement in ML-style generic definitions with multiple type parameters
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKBbBBXANosAqgC4CWOAzgDoBOlAdjfQPQtgCyAMgLTlECeeMAEMcJIeTCwAZuyEAHRnX5z8ACgDkAa0R8ANGHUA3EVkQBKMOnlgA7gAtEdA9r5g0cdHKHUS5WE4BeWTkAHi0dfSMTRAA%2BSF1wKHJEIlI6AHNyVDAAbUSIUAgiqAASZWQ0SBIlePyiyBLjHFNsvOLigAYE9vqASTokJQBlEgAvZG6esAAWOogAXTqAX0mCudLy7Kqa1frG6Nb1iABGXeLINiEADw5qxA5HdKI7WqmwY4AmDvXF4pW6wrtBqbSoAI1gsBwryB%2B2aFVyRzAHzO9SGiHQJDgOH8AEEiABRAYAeSktzoE0RUhEyR%2By12gPOZT4Ki24Mh0MZTRaaDaUwAzCiIJAhl5oIgAEKIKSwaiIAAK3iE6BSiGoHPaRGoplpf3p62BzPhkDZUJRDS58N5PWmgqgIqEYsl0tlHFgNlV0HEiH6hlgntI-nVxU12p6vyK-2KDL2IKgJqDpQthzeAFZbcLRRKpTLEAQ5CpqJ7kj6-UIA3QExAqRREDqI3qegaWWCIab9bDuQi3gA2dP2x3Z2UAYRw4nIQ-8vC10CIMsrYGrNLDdIB7djxtbCfNBx5iIA7H3M06c2x0aDVfPF7Xl7rV42mc245uzR3LYiAByHh1Z52ICfYitbSvOsIEjIpoyFB8jXjF8k13N4AE4vzFHEpCIVUJ3QKx5xDa92nDUCGxhdcYPbOCuymY4ukRDNv2PWU0QxLFAyA6k8OKAiwDAtZ7xI58yJ3CiemOU4aP7RBUPQ6hGMxSEWMRXCQK4ojGT49lYMEq12k%2BdN%2BkGIgiToAAVag%2BAAdRIZ5LzYpTuLACCNkNVl%2BPvcitOKY4BTEzMcWoWAsAGAARRBRAxKScK1diik4uyHKbI1qiILdX2TSibRoy4rl6KQjIcOh8RrIY7BlIgLPgKzbWmb4b3rO9iKcypEuStzEWONMMuufopBIK4iQLMsZXxK45FlchyBIeTU2q-CVyjNcGqgJqNLhVLhN7DqrgAJUQOBqHgMqKsRKrbJUmMFu2JLls7dyimOA8Nu23b4AAOSwdBz2oElenQ9AsltY4Trq1TzvQXBSFEckADEZSsVJVSMpyrrfN5jk-GjHpleA2DBsg7mh6hYfC9NoDsRUZ1VAB9GwSHKl5Abm3jzqWgSVvgyikI23zqCEPgiWoW5eAOl5KumjjZvA%2BbHwu5rNMRL500yrmeb5gWiFe97VS%2Bn6-ta%2BmJcZqXQZwcG8ZhsspIRlkkdW9oPlEt5ICV3n%2Bd8IhseN3GobNuG1WJ0nufJ6gqZpqy9Z4%2BqpeZ1zZbeD5kQ2gA1aJxWqeBqnSIX52OmrCKBs7I52FnrrlryHcyyGApnCa6BTgZ08zkWw-syWEsL6PWaE230rL65AtgIgAHEUiGkbEDG6uG8RFNRei8Xw%2BBqXSPb4vY-asucYhiVsWgTRxQD7QiHIQyhiVP9ITewDKRsnPlLzyC1LbZfkamD51od57EBsTfJSId1HEtxAwVup0EstXHEAxTwa2oDrN4wEb6xRbs5dSRdn49A%2BPdB2ABpRAiA5DZVyo4foJ9lRkgpLA6%2BM1bwMwjq3S6KCbbFA%2BGjHuVwCoYnIHYSebxs6UNqtQhe0EXIwharHDmDsRgZDwL5dIb1HClUQKCNgCAyFTDgbw3O-D86COQU-BhRQ%2BTUQdjiUQ6Q6AVzoFXfwIxTFliwLKIysA9JyLLNXayNYm5xSgkgx%2BwiY78ntlMR2xtVR0BcYYe4-h0iQI%2BkA6ooDJxuKXOo2%2Bmj77nSXr4juN0IB8njuvD2pCOBKlBPAIQI5YDjQyB-L%2BdxElRQWHPZuBttE%2BM5H4nofJS6BMCr4IQoI8BsN8HYIYfAlDXDqR4xBlRHDwBJKQoY-A8AyyyYiPk3dAmElmaSWpxNqA4CkJASZzTvHLJXvyNegTsG4KcUofoe8hAWOFlfdx8DTppMXkItpKy3h8jfoE8Uo46CaFIeQXy-kBgf14IgLGG87gj1GuNBJtpFKvLvo5D5OjMlnI6Rg-53h6KAN8NAHwGJQnoXgAQEB-hgrQFHNzcsEzUWpPRS005qD2h8mYYEhZPgZxKKQIy5JSxEjzEmFUcgkNxrZGAksIAA)

#### Code

```fsharp
module Utils

/// ML-style alias of Map
type ('key, 'value) map when 'key : comparison = Map<'key, 'value>
```
            

#### Result

```fsharp
module Utils

/// ML-style alias of Map
type ('key, 'value when 'key: comparison) map = Map<'key, 'value>

```
            
#### Problem description

Using ML-style generic type definitions is an uncommon but supported syntax choice.

To use multiple type parameters, they're wrapped in parentheses like a tuple. Fantomas puts any generic type constraints inside those parentheses, instead of outside as correct.

#### Extra information

- [x] The formatted result breaks by code.
- [ ] The formatted result gives compiler warnings.
- [x] I or my company would be willing to help fix this.

#### Options

Fantomas Master at 08/11/2021 17:06:14 - 383b729f46ebc01dde282967c68337d5ad55fa17

Default Fantomas configuration

<sub>Did you know that you can ignore files when formatting from fantomas-tool or the FAKE targets by using a [.fantomasignore file](https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md#ignore-files-fantomasignore)?</sub>

### PR DESCRIPTION
**Pull request by josh-degraw**
Closes #1868

### CODE REVIEW

**Comment by nojaf** [comment]
Thanks! Could you add a changelog entry, please?

**Comment by josh-degraw** [comment]
Whoops, my bad. Done.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## Unreleased
+
+### Fixed
+*  Wrong generic constraint placement in ML-style generic definitions with multiple type parameters [#1868](https://github.com/fsprojects/fantomas/issues/1868)
```
Comment (nojaf) [inline CHANGELOG.md:6]: Ok, nitpicking here, but notice that you have two spaces after the `*`.
All the others have one.
And please the line with a `.`.

**Review by nojaf — CHANGES_REQUESTED** [review]
Almost there

File: CHANGELOG.md (hunk 1 above)
Comment (josh-degraw) [reply CHANGELOG.md]: Done

**Review by nojaf — APPROVED** [review]
Thanks!

**Pull request merged**